src/cdk/stepper/stepper.ts

```diff
@@ -0,0 +1,256 @@
+  TemplateRef,
+  ViewEncapsulation
+} from '@angular/core';
+import {LEFT_ARROW, RIGHT_ARROW, ENTER, SPACE} from '@angular/cdk/keycodes';
+import {CdkStepLabel} from './step-label';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {AbstractControl} from '@angular/forms';
+
+/** Used to generate unique ID for each stepper component. */
+let nextId = 0;
+
+/**
+ * Position state of the content of each step in stepper that is used for transitioning
+ * the content into correct position upon step selection change.
+ */
+export type StepContentPositionState = 'previous' | 'current' | 'next';
+
+/** Change event emitted on selection changes. */
+export class StepperSelectionEvent {
+  /** Index of the step now selected. */
+  selectedIndex: number;
+
+  /** Index of the step previously selected. */
+  previouslySelectedIndex: number;
+
+  /** The step instance now selected. */
+  selectedStep: CdkStep;
+
+  /** The step instance previously selected. */
+  previouslySelectedStep: CdkStep;
+}
+
+@Component({
+  selector: 'cdk-step',
+  templateUrl: 'step.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class CdkStep {
+  /** Template for step label if it exists. */
+  @ContentChild(CdkStepLabel) stepLabel: CdkStepLabel;
+
+  /** Template for step content. */
+  @ViewChild(TemplateRef) content: TemplateRef<any>;
+
+  /** The top level abstract control of the step. */
+  @Input()
+  get stepControl() { return this._stepControl; }
+  set stepControl(control: AbstractControl) {
+    this._stepControl = control;
+  }
+  private _stepControl: AbstractControl;
+
+  /** Whether user has seen the expanded step content or not . */
+  interacted = false;
+
+  /** Label of the step. */
+  @Input()
+  label: string;
+
+  @Input()
+  get editable() { return this._editable; }
+  set editable(value: any) {
+    this._editable = coerceBooleanProperty(value);
+  }
+  private _editable = true;
+
+  /** Whether the completion of step is optional or not. */
+  @Input()
+  get optional() { return this._optional; }
+  set optional(value: any) {
+    this._optional = coerceBooleanProperty(value);
+  }
+  private _optional = false;
+
+  /** Return whether step is completed or not. */
+  @Input()
+  get completed() {
+    return this._customCompleted == null ? this._defaultCompleted : this._customCompleted;
+  }
+  set completed(value: any) {
+    this._customCompleted = coerceBooleanProperty(value);
+  }
+  private _customCompleted: boolean | null = null;
+
+  private get _defaultCompleted() {
+    return this._stepControl ? this._stepControl.valid && this.interacted : this.interacted;
+  }
+
+  constructor(private _stepper: CdkStepper) { }
+
+  /** Selects this step component. */
+  select(): void {
+    this._stepper.selected = this;
+  }
+}
+
+@Directive({
+  selector: '[cdkStepper]',
+})
+export class CdkStepper {
+  /** The list of step components that the stepper is holding. */
+  @ContentChildren(CdkStep) _steps: QueryList<CdkStep>;
```

Comment:
Anything that's a `@ContentChild(ren)` or `@ViewChild(ren)` as well as anything referenced in the template must be public, they start with `_` to indicate they're internal